IdentityCore/src/controllers/MSIDSilentController.m

```diff
@@ -118,33 +119,89 @@ - (void)acquireTokenWithRequest:(MSIDSilentTokenRequest *)request
         if (result || !self.fallbackController)
         {
 #if !EXCLUDE_FROM_MSALCPP
-            MSIDTelemetryAPIEvent *telemetryEvent = [self telemetryAPIEvent];
-            [telemetryEvent setUserInformation:result.account];
-            [telemetryEvent setIsExtendedLifeTimeToken:result.extendedLifeTimeToken ? MSID_TELEMETRY_VALUE_YES : MSID_TELEMETRY_VALUE_NO];
-            [self stopTelemetryEvent:telemetryEvent error:error];
+            [self addTelemetryEvent:result error:error];
 #endif
             self.currentRequest = nil;
-            
             completionBlock(result, error);
             return;
         }
 
-        self.currentRequest = nil;
         MSIDRequestCompletionBlock completionBlockWrapper = ^(MSIDTokenResult *ssoResult, NSError *ssoError)
         {
             // We don't have any meaningful information from fallback controller (edge case of SSO error) so we use the local controller result earlier
-            if (!ssoResult && (ssoError.code == MSIDErrorSSOExtensionUnexpectedError))
-            {
-                completionBlock(result, error);
-            }
-            else
+            
+            // If ssoError presented already, skip broker when came back from local RT fallback
+            self.ssoError = ssoError;
+            if (!ssoResult && (self.requestParameters.allowGettingAccessTokenWithRefreshToken || ssoError.code == MSIDErrorSSOExtensionUnexpectedError))
             {
-                completionBlock(ssoResult, ssoError);
+                // Skip duplicate local cache lookups
+                request.forceRefresh = YES;
+                [request executeRequestWithCompletion:^(MSIDTokenResult *localRtResults, NSError *localRtError)
+                {
+#if !EXCLUDE_FROM_MSALCPP
+                    [self addTelemetryEvent:localRtResults error:localRtError];
+#endif
+                    self.currentRequest = nil;
+                    [self completionHandler:localRtResults
+                                  ssoResult:nil
+                                      error:localRtError
+                                   ssoError:self.ssoError
+                            completionBlock:completionBlock];
+                }];
+                return;
             }
+            
+            self.currentRequest = nil;
+            [self completionHandler:nil
+                          ssoResult:ssoResult
+                              error:error
+                           ssoError:self.ssoError
+                    completionBlock:completionBlock];
         };
 
         [self.fallbackController acquireToken:completionBlockWrapper];
     }];
 }
 
+- (void)addTelemetryEvent:(MSIDTokenResult *)result error:(NSError *)error
```

Comment:
Should all of this be excluded from CPP?

Reply:
The logic was just extracted out, yes, the original logic is to exclude all from CPP. I am working on a different PR to get process added in xplat layer here to track sso ext usage:
https://github.com/AzureAD/microsoft-authentication-library-for-cpp

Reply:
I mean, should this code have EXCLUDE_CPP macro around as well?

Reply:
Updated